packages/todo-example/client-urql/src/createUrgqlClient.tsx

```diff
@@ -0,0 +1,81 @@
+import { SocketIOGraphQLClient } from "@n1ru4l/socket-io-graphql-client";
+import {
+  createClient,
+  Exchange,
+  makeErrorResult,
+  OperationResult,
+  Operation,
+  makeResult,
+} from "urql";
+import { pipe, make, mergeMap, share, filter, takeUntil, Source } from "wonka";
+import { print } from "graphql";
+
+export const createUrqlClient = (networkInterface: SocketIOGraphQLClient) => {
+  const socketIOGraphQLUrglExchange: Exchange = ({ client }) => {
+    const createSubscriptionSource = (
+      operation: Operation
+    ): Source<OperationResult> => {
+      const observableish = networkInterface.execute({
+        operation: print(operation.query),
+        variables: operation.variables,
+      });
+
+      return make<OperationResult>(({ next, complete }) => {
+        let isComplete = false;
+        let sub: null | { unsubscribe: () => void } = null;
+
+        sub = observableish.subscribe({
+          next: (result) => next(makeResult(operation, result)),
+          error: (err) => next(makeErrorResult(operation, err)),
+          complete: () => {
+            if (!isComplete) {
+              isComplete = true;
+              if (operation.operationName === "subscription") {
+                client.reexecuteOperation({
+                  ...operation,
+                  operationName: "teardown",
+                });
+              }
+
+              complete();
+            }
+          },
+        });
+
+        return () => {
+          isComplete = true;
+          if (sub) sub.unsubscribe();
+        };
+      });
+    };
+
+    return (operations$) => {
+      const sharedOps$ = share(operations$);
+      return pipe(
+        sharedOps$,
+        mergeMap((operation) => {
+          console.log(
```

Comment:
For some reason, this handler is called multiple times.

![image](https://user-images.githubusercontent.com/14338007/95685948-e06a8300-0bf2-11eb-8832-72ee52904237.png)